server/routes/api/representatives.js

```diff
@@ -89,114 +90,169 @@ router.get('/:searchText', async (req, res) => {
     return
   }
 
-  try {
-    const params = {
-      order: 'district_type', // https://cicero.azavea.com/docs/#order-by-district-type
-      sort: 'asc',
-      max: 200,
-      format: 'json',
-      key: CICERO_API_KEY
-    }
-    // add ZIP code search as parameter.
-    // NOTE: we check it first, because, as of now, the regex use to validate ZIP codes is stricter and more accurate
-    if (isValidZIPcode) {
-      params.search_postal = searchText
-      params.search_country = 'US'
-    }
-    // add street address search as parameter
-    else if (isValidAddress) {
-      params.search_address = streetAddress
-    }
+  const params = {
+    order: 'district_type', // https://cicero.azavea.com/docs/#order-by-district-type
+    sort: 'asc',
+    max: 200,
+    format: 'json',
+    key: CICERO_API_KEY
+  }
 
-    if (filter != null) {
-      params.district_type = JURISDICTION_FILTER_MAP[filter]
-    }
+  // add ZIP code search as parameter.
+  // NOTE: we check it first, because, as of now, the regex use to validate ZIP codes is stricter and more accurate
+  if (isValidZIPcode) {
+    params.search_postal = searchText
+    params.search_country = 'US'
+  }
+  // add street address search as parameter
+  else if (isValidAddress) {
+    params.search_address = streetAddress
+  }
 
-    const {
-      data: { response }
-      //cached
-    } = await axios.get('https://cicero.azavea.com/v3.1/official', {
+  const [received, error] = await getData(
+    'https://cicero.azavea.com/v3.1/official', // url
+    {
       params,
       paramsSerializer: (params) =>
         qs.stringify(params, { arrayFormat: 'repeat' }),
-
       cache: {
         // TODO: we are disabling the cache to test results first
         ttl: 1000 * 60 * 60 * 24 * 7 // the time until the cached value is expired in milliseconds: set to 1 week
       }
-    })
-
-    // if you want to check if response was cached, uncomment: this is a way to track the issue
-    // console.log('isCached:', cached)
-
-    const { errors, results } = response
```

Comment:
You may need to rethink what is happening here. In my local testing, response is undefined and attempting to search crashes the server.

Reply:
This is the code that was deleted / modified. And you are correct, and this is what I've experienced repeatedly while working on `DonateMoney.vue` when the `CICERO_API_KEY` does not work. This is likely what prompted me to refactor `representatives.js` and lump the changes to this PR even when this is technically not within the same issue. 

Did you add `NODE_ENV='development'` in your `.env` file? (I have mine as `.env.dev`.) I've added this particular way to handle the error, but am definitely open to feedback. This is specifically to avoid crashing the server when developing. We can further discuss what should happen if the same request fails in production (e.g., what do we want the client to see? Do we want to redirect to a different page? Or do we want to render a message based on the error response sent back from the server?)... But I did stop myself there. 🐢 

I'll point to my changes on lines 131-179 here, and make some comments there:

https://github.com/ProgramEquity/amplify/blob/1ea6640c76fca1390c19c0672c299b74faa40058/server/routes/api/representatives.js#L131-L179

Reply:
I just tried it with Param's last update, but it doesn't seem to be working for me. I wonder why I'd be an exception though. Wouldn't everyone be having the same issue?

Despite having issues here, I'm still willing to revert these changes.

Reply:
@yoyoyojoe @DietBepis1 For the `NODE_ENV` part, we can plan setting that variable in the `dev` script under #430. I'd already known we'll come across this at some point, haha.

@yoyoyojoe You are right, there is an issue with the key and we're talking about it. Thank you for reporting, I'll make sure we get it fixed as soon as possible. Till then you can use your own or Glenn's trial key.